lib/mongoid/criteria/queryable/smash.rb

```diff
@@ -74,8 +74,12 @@ def [](key)
         # @return [ String ] The normalized key.
         #
         # @since 1.0.0
-        def normalized_key(name, serializer)
-          serializer && serializer.localized? ? "#{name}.#{::I18n.locale}" : name
+        def normalized_key(name, serializer, opts = {})
+          if serializer && serializer.localized? && opts[:localize] != false
```

Comment:
I strongly believe we should change the default behavior, and announce it as a breaking change. It can be a separate PR if needed. The `:localize` option is almost never what anyone would want to use, you'd only want to use it in explicit performance optimization scenarios.